### PR TITLE
Add `int64` type `multiples` support for `tf.tile`

### PR DESCRIPTION
**Pull request by yongtang**
In the doc of `tf.tile` (tf.tile.__doc__) both `int32` and `int64` are supported for `multiples`. However, the kernel for `int64` is not registered yet.

This fix adds the support of `int64` `multiples` so that the behavior matches the description of the docs.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

### CODE REVIEW

File: tensorflow/core/kernels/tile_functor.h [hunk 1]
```diff
@@ -29,13 +30,13 @@ namespace internal {
 template <typename Device, typename T>
 void TileSimple(const Device& d, Tensor* out, const Tensor& in);
 
-template <typename Device, typename T, int NDIM>
+template <typename Device, typename Tmultiples, typename T, int NDIM>
```
Comment (vrv) [inline tensorflow/core/kernels/tile_functor.h]: Minor nit: it seems like T should come before Tmultiples since we usually put the 'main datatype' first and then the auxilliary one.

File: tensorflow/python/kernel_tests/shape_ops_test.py [hunk 2]
```diff
@@ -411,14 +411,17 @@ def testScalar(self):
...
   def testSimple(self):
-    with self.test_session():
-      inp = np.random.rand(4, 1).astype(np.float32)
-      a = constant_op.constant(inp)
-      tiled = array_ops.tile(a, [1, 4])
-      result = tiled.eval()
-    self.assertEqual(result.shape, (4, 4))
-    self.assertEqual([4, 4], tiled.get_shape())
-    self.assertTrue((result == np.tile(inp, (1, 4))).all())
+    for use_gpu in False, True:
```
Comment (vrv) [inline tensorflow/python/kernel_tests/shape_ops_test.py]: You can just do use_gpu=True; if a GPU is not available the test will run on CPU, and our test infrastructure runs all tests on CPU and GPU machines.

**Review by vrv — COMMENTED** [review]
Overall looks good, one just one minor ordering suggestion

File: tensorflow/core/kernels/tile_functor.h (hunk 1 above)
Comment (yongtang) [reply tensorflow/core/kernels/tile_functor.h]: Thanks @vrv, the order of the typename has been rearranged.

File: tensorflow/python/kernel_tests/shape_ops_test.py (hunk 2 above)
Comment (yongtang) [reply tensorflow/python/kernel_tests/shape_ops_test.py]: Thanks @vrv. Done.

**Comment by yongtang** [comment]
Thanks @vrv for the review. The PR has been updated. Please take a look.

**Comment by vrv** [comment]
@tensorflow-jenkins test this please

**Review by vrv — APPROVED** [review]

File: tensorflow/core/kernels/tile_ops.cc [hunk 3]
```diff
@@ -505,127 +508,73 @@ REGISTER_KERNEL_BUILDER(Name("Tile")
...
+  REGISTER_KERNEL_BUILDER(Name("Tile")                             \
+                              .Device(DEVICE_GPU)                  \
+                              .TypeConstraint<type>("T")           \
+                              .TypeConstraint<int64>("Tmultiples") \
+                              .HostMemory("multiples"),            \
+                          TileOp<GPUDevice, int64>);               \
+  REGISTER_KERNEL_BUILDER(Name("TileGrad")                         \
+                              .Device(DEVICE_GPU)                  \
+                              .TypeConstraint<type>("T")           \
+                              .TypeConstraint<int32>("Tmultiples") \
```
Comment (vrv) [inline tensorflow/core/kernels/tile_ops.cc]: should we also add the int64 multiples kernel for TileGrad too?  I think we can do that in a separate PR, but it probably makes sense to do this for completeness.

**Review by vrv — APPROVED** [review]

File: tensorflow/core/kernels/tile_ops.cc (hunk 3 above)
Comment (yongtang) [reply tensorflow/core/kernels/tile_ops.cc]: @vrv Again thanks for the review and sorry for the `TileGrad`. Let me add a PR shortly.
